### PR TITLE
Use shutter device in I24 serial plans

### LINKED ISSUE #110: I24ssx: use shutter device in plans
Use an ophyd device to open and close the hutch shutter at beginning at end of data collection.
First, the device in dodal needs to be created see [dodal#611](https://github.com/DiamondLightSource/dodal/issues/611)

### PR DESCRIPTION
**Pull request by noemifrisina**
Closes #110 

Needs [Dodal#623](https://github.com/DiamondLightSource/dodal/pull/623)

### CODE REVIEW

File: src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py [hunk 1]
```diff
@@ -192,12 +194,9 @@ def run_extruder_plan(
...
 
     # Setting up the beamline
-    caput("BL24I-PS-SHTR-01:CON", "Reset")
-    logger.debug("Reset hutch shutter sleep for 1sec")
-    sleep(1.0)
-    caput("BL24I-PS-SHTR-01:CON", "Open")
-    logger.debug("Open hutch shutter sleep for 2sec")
-    sleep(2.0)
+    logger.debug("Open hutch shutter")
+    # Before Open, it Resets
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py:198]: Should: This is an implementation detail of the shutter, I don't think we care about it from the plan level. I think we can remove the comment.

File: src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py [hunk 2]
```diff
@@ -192,12 +194,9 @@ def run_extruder_plan(
     parameters = ExtruderParameters.from_file(PARAM_FILE_PATH / PARAM_FILE_NAME)
 
     # Setting up the beamline
-    caput("BL24I-PS-SHTR-01:CON", "Reset")
-    logger.debug("Reset hutch shutter sleep for 1sec")
-    sleep(1.0)
-    caput("BL24I-PS-SHTR-01:CON", "Open")
-    logger.debug("Open hutch shutter sleep for 2sec")
-    sleep(2.0)
+    logger.debug("Open hutch shutter")
```
Comment (DominicOram) [inline src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py:197]: Should: Why is opening the shutter logged at debug and closing logged at info? Feels like they should be logged at the same level?

**Review by DominicOram — CHANGES_REQUESTED** [review]
Great, thank you! Couple of comments in code. Additionally it would be good if we asserted that the shutter does get open/closed in some of the tests

File: src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py (hunk 2 above)
Comment (noemifrisina) [reply src/mx_bluesky/I24/serial/extruder/i24ssx_Extruder_Collect_py3v2.py:197]: Yeah, should both be at debug. Must have missed the other one, thanks

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**